vault/vault.go

```diff
@@ -151,20 +153,26 @@ func (v *vaultSecrets) String() string {
 	return Name
 }
 
-func (v *vaultSecrets) keyPath(secretID, namespace string) string {
-	if v.isKvBackendV2 {
-		return path.Join(namespace, v.backendPath, kvDataKey, secretID)
+func (v *vaultSecrets) keyPath(secretID, namespace string) keyPath {
+	if namespace == "" {
+		namespace = v.namespace
+	}
+	return keyPath{
+		backendPath: v.backendPath,
+		isBackendV2: v.isKvBackendV2,
+		namespace:   namespace,
+		secretID:    secretID,
 	}
-	return path.Join(namespace, v.backendPath, secretID)
 }
 
 func (v *vaultSecrets) GetSecret(
 	secretID string,
 	keyContext map[string]string,
 ) (map[string]interface{}, error) {
-	secretValue, err := v.read(v.keyPath(secretID, keyContext[secrets.KeyVaultNamespace]))
+	key := v.keyPath(secretID, keyContext[secrets.KeyVaultNamespace])
+	secretValue, err := v.read(key.Path())
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%s: %s", key, err)
```

Comment:
nit: add a prefix to the error: `fmt.Errorf("failed to get secret %s: %s", key, err)`
and at other APIs as well.

Reply:
updated